### PR TITLE
Prefer BP_HTTPD_VERSION to buildpack.yml configuration

### LINKED ISSUE #168: Implement RFC 0001: BP_HTTPD_VERSION to request version
See [RFC 0001: Buildpack.yml to Environment Variables](https://github.com/paketo-buildpacks/httpd/blob/main/rfcs/0001-buildpack-yml-to-env-var.md). 

The buildpack should support:
* Configuration using `$BP_HTTPD_VERSION `
* The buildpack should support both configuration options with environment variables taking priority
* The buildpack should detect whether the application has a buildpack.yml and print a message warning the deprecation of its use.

### PR DESCRIPTION
**Pull request by arjun024**
Resolves #168 
See issue for more details

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.

### CODE REVIEW

File: build_test.go [hunk 1]
```diff
@@ -28,8 +29,10 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 		cnbPath    string
 		timestamp  string
 
+		entryResolver     *fakes.EntryResolver
```
Comment (thitch97) [inline build_test.go:32]: Looks like this just needs `go generate` run

File: build_test.go [hunk 2]
```diff
@@ -73,15 +88,23 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
...
 			Stack:      "some-stack",
 			Plan: packit.BuildpackPlan{
 				Entries: []packit.BuildpackPlanEntry{
 					{
-						Name:     "httpd",
-						Metadata: map[string]interface{}{"launch": true},
+						Name: "httpd",
+						Metadata: map[string]interface{}{
+							"version-source": "BP_HTTPD_VERSION",
+							"version":        "some-entry-version",
```
Comment (thitch97) [inline build_test.go:105]: should this be "some-env-var-version" as on line 144?

File: build_test.go [hunk 3]
```diff
@@ -184,7 +213,7 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 
 			Expect(dependencyService.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbPath, "buildpack.toml")))
 			Expect(dependencyService.ResolveCall.Receives.Name).To(Equal("httpd"))
-			Expect(dependencyService.ResolveCall.Receives.Version).To(Equal("2.4.*"))
+			Expect(dependencyService.ResolveCall.Receives.Version).To(Equal("some-env-var-version"))
```
Comment (thitch97) [inline build_test.go:216]: Same as line 105?

**Review by thitch97 — CHANGES_REQUESTED** [review]

File: build_test.go (hunk 1 above)
Comment (sophiewigmore) [reply build_test.go:32]: Not sure how the tests passed locally... maybe we forgot to stage the fake directory SMH. Thanks!

**Review by thitch97 — APPROVED** [review]

**Pull request merged**